controllers/social.ts

```diff
@@ -0,0 +1,244 @@
+import prisma from "../prisma/client";
+import asyncHandler from "express-async-handler";
+import { Prisma } from "@prisma/client";
+import db from "../library/db";
+import { SplitContentTypes } from "../library/userHelper";
+import log from "loglevel";
+import {
+  getContentFromId,
+  getParentContentTypeAndId,
+} from "../library/content";
+
+type ActivityType = "playlistCreate" | "zap" | "updatePlaylist";
+interface ActivityItem {
+  picture: string;
+  name: string;
+  userId: string;
+  pubkey: string;
+  type: ActivityType;
+  message?: string;
+  zapAmount?: number;
+  timestamp: string;
+  contentId: string;
+  contentTitle: string;
+  contentType: SplitContentTypes | "playlist";
+  contentArtwork: string[];
+  parentContentId?: string;
+  parentContentTitle?: string;
+  parentContentType?: string;
+}
+
+const getActivity = async (
+  pubkeys: string[],
+  limit: number,
+  offset: number = 0
+) => {
+  const threeMonthsAgo = new Date();
+  threeMonthsAgo.setMonth(threeMonthsAgo.getMonth() - 3);
+  const filterDate = threeMonthsAgo
+    .toISOString()
+    .slice(0, 19)
+    .replace("T", " ");
+
+  const createdPlaylists = await db
+    .knex("playlist")
+    .whereIn("user_id", pubkeys)
+    .join("npub", "playlist.user_id", "npub.public_hex")
+    .select(
+      "playlist.id as id",
+      "playlist.user_id as user_id",
+      "playlist.title as title",
+      "playlist.updated_at as updated_at",
+      "npub.metadata as metadata"
+    )
+    .orderBy("playlist.updated_at", "desc")
+    .where("playlist.updated_at", ">", filterDate);
+
+  const createdPlaylistActivity: ActivityItem[] = [];
+  for (const playlist of createdPlaylists) {
+    const tracks = await db
+      .knex("playlist_track")
+      .select(
+        "track_info.artwork_url as artworkUrl",
+        "playlist_track.order_int as order"
+      )
+      .join("track_info", "track_info.id", "=", "playlist_track.track_id")
+      .where("playlist_track.playlist_id", playlist.id)
+      .orderBy("playlist_track.order_int", "asc");
+
+    if (tracks.length !== 0) {
+      createdPlaylistActivity.push({
+        picture: playlist.metadata?.picture,
+        name: playlist.metadata?.name,
+        userId: playlist.user_id,
+        pubkey: playlist.user_id,
+        type: "playlistCreate",
+        timestamp: playlist.updated_at,
+        contentId: playlist.id,
+        contentTitle: playlist.title,
+        contentType: "playlist",
+        contentArtwork: tracks.map((track) => track.artworkUrl),
+      });
+    }
+  }
+  const zaps = await db
+    .knex("amp")
+    .whereIn("amp.user_id", pubkeys)
+    .andWhere("amp.comment", true)
```

Comment:
Could add an additional filter here for type 7, which is zaps.

```suggestion
    .andWhere("amp.comment", true)
    .andWhere("amp.type", 7)
```